packages/datadog-plugin-connect/test/index.spec.js

```diff
@@ -0,0 +1,587 @@
+          })
+
+          app.use('/user', (req, res) => {
+            res.end()
+
+            try {
+              expect(tracer.scope().active()).to.not.be.null.and.not.equal(span)
+              done()
+            } catch (e) {
+              done(e)
+            }
+          })
+
+          getPort().then(port => {
+            appListener = http.createServer(app).listen(port, 'localhost', () => {
+              axios.get(`http://localhost:${port}/user`)
+                .catch(done)
+            })
+          })
+        })
+
+        it('should only include paths for routes that matched', done => {
+          const app = connect()
+
+          app.use('/foo', (req, res, next) => next())
+          app.use('/app', (req, res) => res.end())
+          app.use('/bar', (req, res, next) => next())
+
+          getPort().then(port => {
+            agent
+              .use(traces => {
+                const spans = sort(traces[0])
+
+                expect(spans[0]).to.have.property('resource', 'GET /app')
+              })
+              .then(done)
+              .catch(done)
+
+            appListener = http.createServer(app).listen(port, 'localhost', () => {
+              axios.get(`http://localhost:${port}/app/user/123`)
+                .catch(done)
+            })
+          })
+        })
+
+        it('should extract its parent span from the headers', done => {
+          const app = connect()
+
+          app.use('/user', (req, res) => {
+            res.end()
+          })
+
+          getPort().then(port => {
+            agent.use(traces => {
+              const spans = sort(traces[0])
+
+              expect(spans[0].trace_id.toString()).to.equal('1234')
+              expect(spans[0].parent_id.toString()).to.equal('5678')
+            })
+              .then(done)
+              .catch(done)
+
+            appListener = http.createServer(app).listen(port, 'localhost', () => {
+              axios
+                .get(`http://localhost:${port}/user`, {
+                  headers: {
+                    'x-datadog-trace-id': '1234',
+                    'x-datadog-parent-id': '5678',
+                    'ot-baggage-foo': 'bar'
+                  }
+                })
+                .catch(done)
+            })
+          })
+        })
+
+        it('should handle error status codes', done => {
+          const app = connect()
+
+          app.use((req, res, next) => {
+            next()
+          })
+
+          app.use('/user', (req, res) => {
+            res.statusCode = 500
+            res.end()
+          })
+
+          getPort().then(port => {
+            agent.use(traces => {
+              const spans = sort(traces[0])
+
+              expect(spans[0]).to.have.property('error', 1)
+              expect(spans[0]).to.have.property('resource', 'GET /user')
+              expect(spans[0].meta).to.have.property('http.status_code', '500')
+
+              done()
```

Comment:
Let's call `done` in `then` instead of calling `done()` ourselves. Same for L414.